enhancements/network/ovn-secondary-networks.md

```diff
@@ -0,0 +1,240 @@
+    // Subnet is a RFC 4632/4291-style string that represents an IP address and prefix length in CIDR notation
+    Subnet                  string `json:"subnet"`
+    HasExternalConnectivity bool   `json:"hasExternalConnectivity,omitempty"`
+}
+```
+
+A controller will watch out for the creation / deletion of these
+`OVNSecondaryNetwork`s, and will render and provision the corresponding
+`NetworkAttachmentDefinition`s.
+
+#### Network-attachment-definition provisioning
+When a new network attachment is provisioned (or removed), the corresponding
+controller will create a logical switch (named
+`secondary_<namespace>_<networkName>`), which will assign IP addresses in the
+configured range (defined by the `OVNSecondaryNetwork`).
+
+A removed `Network-Attachment-definition` with present attachment will
+trigger a reconciliation loop, which will cause the net-attach-def entity to be
+reprovisioned; if the network attachment definition is removed and the
+corresponding OVN logical switch does not have any logical ports attached
+triggers the logical switch to be deleted.
+
+#### Adding / removing a pod using an OVN secondary network
+Adding / removing a pod will lead to a logical switch port to be created,
+attached to the corresponding logical switch.
+
+A static MAC / IP address can be requested for the pods by specifying those in
+the annotations field. The requested IP address **must** be in range of the
+corresponding `OVNSecondaryNetwork`.
+
+### Risks and Mitigations
+
+## Design Details
+
+### Net-attach-def samples
+This is an example of a network attachment definition that will create an
+isolated flat L2 network.
+
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenant-network-1
+  namespace: testns1
+spec:
+  config: '{
+            "cniVersion": "0.3.1",
+            "type": "ovn-kubernetes",
+            "name": "tenant-overlay",
+            "subnet": "10.10.10.0/24",
+        }'
+```
+
+This is an example of a network attachment definition that will create a flat
+L2 network connected to the external network.
+```yaml
+apiVersion: "k8s.cni.cncf.io/v1"
+kind: NetworkAttachmentDefinition
+metadata:
+  name: tenant-network-2
+  namespace: testns1
+spec:
+  config: '{
+            "cniVersion": "0.3.1",
+            "type": "ovn-kubernetes",
+            "name": "tenant-overlay-ext-network",
+            "subnet": "10.10.20.0/24",
+            "has_external_connectivity": true,
+        }'
+```
+
+- a logical switch per network. We will define an external id "owners" field
+  so other ovn related projects can keep away from our stuff (e.g. avoid ovn-k
+  reconciliation loops)
+  - for IPv6 subnets we would need to also create a logical router and
+    logical router port, to be able to send the RAs. Would we want
+    these routers for other scenarios ? I.e. is there any reason to create
+    them for IPv4 subnets ?
+- MTU propagation via dhcp options (ipv4)
```

Comment:
Do we expect DHCP to be available? If so, how does that interact with the static addressing requests in the pod's annotations? What happens if the pod specifies an IP that's already leased by DHCP?